python/salpytools/salpylib.py

```diff
@@ -235,6 +235,8 @@ def subscribe(self):
         self.mgr_ackCommand = getattr(self.mgr, 'ackCommand_{}'.format(self.command))
         # And the code for the SAL__CMD_COMPLETE from the library itself
         self.SAL__CMD_COMPLETE = SALPY_lib.SAL__CMD_COMPLETE
+        self.SAL__CMD_FAILED = SALPY_lib.SAL__CMD_FAILED
+        self.SAL__CMD_NOPERM = SALPY_lib.SAL__CMD_NOPERM
```

Comment:
It seems a little strange to copy all these into the instance when they're constant and won't be changed.  Why not just use them directly from `SALPY_lib`?  (Also, you have `SALPY_lib` as a `subscribe()` method variable, as an instance variable set in `subscribe()` and as an instance variable set in `__init__()`.  There doesn't seem to be a reason for handling this three different ways.)

Reply:
Yes, that was bothering me too and should I have simple use `SALPY_lib` and avoid these shortcut. Thanks.